### DISCUSSION TITLE
Rethink how we should handle themes when the node doesn't support it (e.g. Label3D)

### DISCUSSION THREAD

**Discussion by nathanfranke** [godotengine/godot-proposals#4919] [GUI]
See discussion in https://github.com/godotengine/godot/pull/62139

My proposal:
---

Remove `Sprite3D` and `Label3D` and add a new node that displays any arbitrary GUI in a 3d plane.

Internally, it will work something like this:

```
Viewport
→ GUI nodes
MeshInstance3D
→ QuadMesh
    → Material
        → ViewportTexture
```

(Our new node will either have to extend viewport or do something fancy to actually create this hierarchy)

Question about this, will the ancestor theme still be propagated even through a Viewport?

**Comment by h0lley** [comment]
are there common enough use-cases for GUI other than Label in 3D space?

**Reply by nathanfranke** [reply]
VR is going to be a huge use case. Would also be useful for anything more complex than a single label (e.g. a hboxcontainer with an icon and a label)

**Reply by AndreaTerenz** [reply]
You could also use this for "floating" UI like markers on targets/objectives/enemies

**Reply by Calinou** [reply]
> You could also use this for "floating" UI like markers on targets/objectives/enemies

Sprite3D can already be used for markers in a 3D world. When used in tandem with Label3D, this covers a lot of use cases already :slightly_smiling_face: 

Progress bars can be emulated using two Sprite3D nodes or two Label3D nodes: https://github.com/godotengine/godot-demo-projects/pull/761

**Reply by AndreaTerenz** [reply]
@Calinou yes but it's a bit limited (cant make a whole UI out of labels and sprites). Thankfully [this demo](https://godotengine.org/asset-library/asset/889) exists
